src/pytest_workflow/util.py

```diff
@@ -211,6 +212,38 @@ def file_md5sum(filepath: Path, block_size=64 * 1024) -> str:
     return hasher.hexdigest()
 
 
+def file_diff(first_filepath: Path, second_filepath: Path,
+              encoding: Optional[str] = None,
+              max_lines: int = 20) -> Optional[str]:
+    """
+    Generates a diff between two files.
+    :param first_filepath: a pathlib.Path to the first file
+    :param second_filepath: a pathlib.Path to the second file
+    :param encoding: a pathlib.Path to the second file.
+    :param max_lines: maximum number of diff lines to include
+    :return: a unified diff of the two files, or None if they are identical
+    """
+    if encoding is None:
+        encoding = sys.getdefaultencoding()
+
+    with first_filepath.open('r', encoding=encoding) as first_file_handler:
```

Comment:
There should be some code here to handle gzipped content, because that is the case for the `contains` and `must_not_contain` fields as well. If that is not handled here too, that will lead to unexpected and inconsistent behavior.